Bump version to 7.1.2 & switch to gha ci for image deployments

adamus1red opened the pull request:
The GHA workflow pulls the version numbers to build from the SOPEL_BRANCH environment variable in the Dockerfile rather than relying on changing it in 3 different places.

Has a matrix build for the different supported python versions with Python 3.9 defaulting to be the "main" release version as the latest tag.

Also fixes #33

Example run of GHA action: [adamus1red/docker-sopel/actions/runs/1042705230](https://github.com/adamus1red/docker-sopel/actions/runs/1042705230)

New built-time secrets 
* IMAGE: name of the image to be tagged with (i.e. sopelirc/sopel)
* DOCKER_USERNAME: Username to login to docker hub with.
* DOCKER_PASSWORD: Access token to authenticate with docker hub.

HumorBaby commented:
oh boy, merging from master?? :cold_sweat:
A lot of this looks good, and since I'm so behind on switch to GHA, I imagine we can use almost everything here… 
_however_, I would like to request the following before a real review:
- remove excess files from PR; I think your current branch has extra CI stuff, e.g., dependabot…??

_then_, I would like to discuss with you (and anyone else that wants to chime in) about:
- splitting out multi-arch from a first-step straight-forward build+label+push. This can keep the multiple python versions as you are doing now. Once this is merged in a nice and clean state, let's 100% get multi-arch done. 

Gotta run, but wanted to share my thoughts without further delay. Let me know what you think

adamus1red commented:
>     * splitting out multi-arch from a first-step straight-forward build+label+push. This can keep the multiple python versions as you are doing now. Once this is merged in a nice and clean state, let's 100% get multi-arch done.

I've split the single build into #42 with single arch build. After it is done I'll add the multi-arch on top.


> I would like to request the following before a real review:
> 
>     * remove excess files from PR; I think your current branch has extra CI stuff, e.g., dependabot…??

I added Dependabot to keep the actions that are used by GHA upto date.
